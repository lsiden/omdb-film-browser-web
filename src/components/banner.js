import React from "react"
import QueryForm from "query-form"
import { OMDB_URL } from "constants"

const headerStyle = {
  background: "linear-gradient(to right, #000040, #3b00ff)",
  padding: "10px",
  color: "white",
}
const titleStyle = {
  fontSize: "18pt",
  marginBottom: "18pt",
}

const linkStyle = {
  color: "white",
  fontSize: "small",
}

const Banner = () => (
  <header style={headerStyle}>
    <h1 style={titleStyle}>
      {"Search Open Movie Database"}&nbsp;
      <a href={OMDB_URL} style={linkStyle}>
        (OMDB)
      </a>
    </h1>
    <QueryForm />
  </header>
)

export default Banner
