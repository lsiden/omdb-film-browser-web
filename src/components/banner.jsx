import React from 'react'

import { headerStyle, titleStyle } from 'style'
import { BANNER_TITLE } from 'omdb_constants'

const AboutLink = () => (
  <a href="/about" style={{ color: 'white' }}>
    about
  </a>
)

const inAboutPath = () => {
  return /about\/?$/.test(window.location.href)
}

export default () => {
  return (
    <header style={headerStyle}>
      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <h1 style={titleStyle}>{BANNER_TITLE}</h1>
        {!inAboutPath() && <AboutLink />}
      </div>
    </header>
  )
}