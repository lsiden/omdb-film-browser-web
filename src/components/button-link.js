import React from "react"
import PropTypes from "prop-types"

const ButtonLink = (props) => <button className="btn btn-link" {...props} />

ButtonLink.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
    .isRequired,
}

export default ButtonLink
