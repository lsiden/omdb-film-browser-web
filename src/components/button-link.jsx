import React from 'react'
import PropTypes from 'prop-types'

const ButtonLink = ({ style, onClick, children }) => (
  /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
  <button
    type="button"
    className="btn btn-link"
    style={style}
  >
    {children}
  </button>
)

ButtonLink.propTypes = {
  style: PropTypes.object,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
}

ButtonLink.defaultProps = {
  style: {},
}

export default ButtonLink
