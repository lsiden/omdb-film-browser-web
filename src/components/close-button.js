import React from "react"
import PropTypes from "prop-types"

import ButtonLink from "components/button-link"

const CLOSE_ICON = "\u2715"

const closeButtonStyle = {
  margin: "8pt",
  fontSize: "20pt",
}

export const CloseButton = ({ onClick }) => (
  <ButtonLink style={closeButtonStyle} onClick={onClick} title="Close">
    {CLOSE_ICON}
  </ButtonLink>
)

CloseButton.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default CloseButton
