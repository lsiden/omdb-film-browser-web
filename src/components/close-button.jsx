import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import ButtonLink from 'components/button-link'

const CLOSE_ICON = '\u2715'

const closeButtonStyle = {
  margin: 20,
  fontSize: 20,
}

const CloseButton = ({ buttonStyle, onClick }) => {
  const style = {
    ...closeButtonStyle,
    ...buttonStyle,
  }
  return (
    <Link to="/">
      <ButtonLink onClick={onClick} style={style} title="Close">
        {CLOSE_ICON}
      </ButtonLink>
    </Link>
  )
}
CloseButton.propTypes = {
  buttonStyle: PropTypes.object,
}
CloseButton.defaultProps = {
  buttonStyle: {},
}

export default CloseButton
