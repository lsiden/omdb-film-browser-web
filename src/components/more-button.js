import React from "react"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import Spinner from "respin"

import { pageFetch } from "actions/remote"

const spinnerStyle = {
  marginLeft: 45,
}
export const PureMoreButton = ({ isFetching, dispatchPageFetch }) =>
  isFetching ? (
    <div style={spinnerStyle}>
      <Spinner />
    </div>
  ) : (
    <button className="more-button" onClick={dispatchPageFetch}>
      More
    </button>
  )

PureMoreButton.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  dispatchPageFetch: PropTypes.func.isRequired,
}

export default connect(
  state => ({
    isFetching: !!state.isFetching,
  }),
  dispatch => ({
    dispatchPageFetch: () => dispatch(pageFetch()),
  })
)(PureMoreButton)
