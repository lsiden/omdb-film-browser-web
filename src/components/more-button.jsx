import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { promiseQueryPageFetch } from 'actions/remote'
import NavButton from 'components/nav-button'

const buttonStyle = {
  marginBottom: 20,
  fontSize: 20,
  color: 'darkgrey',
}

export const _MoreButton = ({ dispatchPageFetch }) => (
  <NavButton onClick={dispatchPageFetch} style={buttonStyle} title="More">
    More
  </NavButton>
)

_MoreButton.propTypes = {
  dispatchPageFetch: PropTypes.func.isRequired,
}

export default connect(
  () => ({}),
  (dispatch) => ({
    dispatchPageFetch: () => dispatch(promiseQueryPageFetch()),
  }),
)(_MoreButton)
