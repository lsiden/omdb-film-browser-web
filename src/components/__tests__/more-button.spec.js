import React from "react"
import TestRenderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'

import { PureMoreButton as MoreButton } from "components/more-button"

const defaultProps = () => ({
  isFetching: false,
  dispatchPageFetch: () => {},
})

const createWrapper = (props = {}) => {
  return TestRenderer.create(<StaticRouter context={{}}>
    <MoreButton {...{ ...defaultProps(), ...props }} />
    </StaticRouter>)
}

it("matches snapshot when not fetching", () => {
  expect(createWrapper({ isFetching: false })).toMatchSnapshot()
})

it("matches snapshot when fetching", () => {
  expect(createWrapper({ isFetching: true })).toMatchSnapshot()
})
