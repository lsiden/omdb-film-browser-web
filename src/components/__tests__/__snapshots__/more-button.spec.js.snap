// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when fetching 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dispatchPageFetch={[Function]}
    isFetching={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown />,
      "style": Object {
        "marginBottom": 20,
        "marginLeft": 45,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown />,
        "style": Object {
          "marginBottom": 20,
          "marginLeft": 45,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`matches snapshot when not fetching 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dispatchPageFetch={[Function]}
    isFetching={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "More",
      "onClick": [Function],
      "style": Object {
        "color": "darkgrey",
        "fontSize": 20,
        "marginBottom": 20,
      },
      "title": "More",
    },
    "ref": null,
    "rendered": "More",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "More",
        "onClick": [Function],
        "style": Object {
          "color": "darkgrey",
          "fontSize": 20,
          "marginBottom": 20,
        },
        "title": "More",
      },
      "ref": null,
      "rendered": "More",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
